jaxrs/src/main/java/org/killbill/billing/jaxrs/resources/SubscriptionResource.java

```diff
@@ -352,10 +351,12 @@ public Response createSubscriptionsWithAddOnsInternal(final List<BulkSubscriptio
 
             final LocalDate resolvedEntitlementDate = toLocalDate(entitlementDate);
```

Comment:
My understanding is that both `entitlementDate` and `billingDate` could be either a `LocalDate` or a `DateTime`: So we should not anymore convert using `toLocalDate`:
* If this is a `DateTime` (i.e we can deserialize the input string into a DateTime), we take it as-is
* If not, then it should be a valid `LocalDate` and we do the gymnastics to convert it into a `DateTime`

Reply:
Yes agreed, I will be making these JAXRS layer changes for all methods ( `create`, `change`, `cancel` ..) in the next round.